### PR TITLE
Refactor find_wildcard_matches() to allow for proper testing

### PR DESCRIPTION
**Pull request by ajor**
Some tests weren't running correctly (add_probes_character_class)
and some were system-dependent (e.g. add_probes_uprobe_wildcard).

### CODE REVIEW

File: tests/bpftrace.cpp [hunk 1]
```diff
@@ -27,6 +27,41 @@ using ::testing::ContainerEq;
...
+        printf("doing ok\n");
+        return myval;
+      });
+
+  ON_CALL(*bpftrace,
+          get_symbols_from_file("/sys/kernel/debug/tracing/available_events"))
+      .WillByDefault([](const std::string &)
+      {
+        std::string tracepoints = "sched:sched_one\n"
+                                  "sched:sched_two\n";
```
Comment (ajor) [inline tests/bpftrace.cpp]: I should add some extra entries here before merging

File: tests/bpftrace.cpp [hunk 2]
```diff
@@ -27,6 +27,41 @@ using ::testing::ContainerEq;
...
+        return std::unique_ptr<std::istream>(new std::istringstream(tracepoints));
+      });
+
+  std::string usyms = "first_open\n"
+                      "second_open\n";
```
Comment (ajor) [inline tests/bpftrace.cpp]: here too

**Comment by ajor** [comment]
Wait for #508 to be merged before merging this. Should also add USDT wildcard tests.

**Comment by ajor** [comment]
This is waiting on `find_usdt_wildcard_matches()` being removed

**Comment by ajor** [comment]
I had to do some nasty hacks in here to make all the related code testable

**Comment by ajor** [comment]
Yep, tests are failing because of that. That's what I meant to say in my comments above but wasn't very clear

File: src/ast/codegen_llvm.cpp [hunk 3]
```diff
@@ -1300,38 +1300,54 @@ void CodegenLLVM::visit(Probe &probe)
...
       {
         case ProbeType::kprobe:
         case ProbeType::kretprobe:
-          matches = bpftrace_.find_wildcard_matches(attach_point->target,
-                                                    attach_point->func,
-                                                    "/sys/kernel/debug/tracing/available_filter_functions");
+        {
+          symbol_stream = bpftrace_.get_symbols_from_file(
+              "/sys/kernel/debug/tracing/available_filter_functions");
+          prefix = "";
```
Comment (ajor) [inline src/ast/codegen_llvm.cpp]: This was deliberate - I don't think kprobes have a target

**Comment by ajor** [comment]
Noticed I'd missed the case where there is no namespace for a USDT probe. I've added a test to make sure that has a wildcard inserted too now. i.e. `usdt:/bin/sh::mytracepoint` is equivalent to `usdt:/bin/sh:*:mytracepoint`

@dalehamel Could you check the new test cases I've added (`bpftrace.add_probes_usdt_wildcard` and `bpftrace.add_probes_usdt_empty_namespace`) to make sure that's how you expect USDT wildcarding to work?

And would someone also be able to run the USDT runtime tests to ensure I've not broken them?

**Comment by dalehamel** [comment]
Will do @ajor, also have a fix for the broken alpine builds locally. Will get to this in the next few days, I have a few house projects I need to wrap up.

File: src/ast/codegen_llvm.cpp [hunk 4]
```diff
@@ -1300,38 +1300,54 @@ void CodegenLLVM::visit(Probe &probe)
...
           break;
+        }
         case ProbeType::usdt:
         {
-          auto usdt_symbol_stream = USDTHelper::probe_stream(bpftrace_.pid_, attach_point->target);
-          matches = bpftrace_.find_usdt_wildcard_matches(attach_point->ns, attach_point->func, usdt_symbol_stream);
+          symbol_stream = bpftrace_.get_symbols_from_usdt(
+              bpftrace_.pid_, attach_point->target);
+          prefix = "";
+          func = attach_point->ns + ":" + attach_point->func;
```
Comment (ajor) [inline src/ast/codegen_llvm.cpp]: Missed this yesterday - I'll do a bit more refactoring so that code doesn't need to be duplicated here

**Pull request merged**